### DISCUSSION TITLE
How te remove (not hide) elements

### DISCUSSION THREAD

**Discussion by GentleGiantSoftware** [AustL/PygameWidgets#49] [Q&A]
Is there a way to remove all elements or a specific element (e.g. a button) instead of only hiding it?

**Comment by slashformotion** [comment]
remove ?

**Comment by marcstober** [comment]
I would like to know how to do this, too. For example, if in a program a user moves between different "screens" I might want to be able to clear things and redraw the screen with different widgets.

**Reply by marcstober** [reply]
I found a way to do this:

```
        widgets = pygame_widgets.WidgetHandler.getWidgets()
        widgets.remove(button1)

```

**Reply by slashformotion** [reply]
that works but we should have a way to do this using the WidgetHandler directly

**Comment by gogopro-dev** [comment]
My recent pull request has been merged, now WidgetHandler does not hold a strong reference to element. That means that you have the absolute control over the widgets life cycle and the widget instantce will be destructed as soon as you hold no strong reference to it.
